### PR TITLE
Add new option to mail collector

### PR DESCRIPTION
**Pull request by stonebuzz**
This PR add new option to mail collector to add or not cc user as ticket observer 

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: tests/imap/MailCollector.php [hunk 1]
```diff
@@ -199,16 +200,18 @@ private function doConnect() {
...
-         'server_cert'  => '/novalidate-cert'
+         'name'               => 'testuser',
+         'login'              => 'testuser',
+         'is_active'          => true,
+         'passwd'             => 'applesauce',
+         'mail_server'        => '127.0.0.1',
+         'server_type'        => '/imap',
+         'server_port'        => 143,
+         'server_ssl'         => '',
+         'add_cc_to_observer' => 1, //add ccuser as observer in ticket
```
Comment (trasher) [inline tests/imap/MailCollector.php]: Relation ID should not be hardcoded

Comment (trasher) [reply tests/imap/MailCollector.php]: (I wonder if this is not the cause of tests failure)

File: install/update_94_95.php [hunk 2]
```diff
@@ -941,6 +941,10 @@ function update94to95() {
       }
    }
 
+     /** Add new option to mailcollector */
+     $migration->addField("glpi_mailcollectors", "add_cc_to_observer", "tinyint(1) NOT NULL DEFAULT '0'");
```
Comment (trasher) [inline install/update_94_95.php]: Maybe better to use the predefined type boolean?

File: inc/mailcollector.class.php [hunk 3]
```diff
@@ -952,11 +957,11 @@ function buildTicket($uid, \Zend\Mail\Storage\Message $message, $options = []) {
 
       // Add to and cc as additional observer if user found
       $ccs = $headers['ccs'];
-      if (is_array($ccs) && count($ccs)) {
+      if (is_array($ccs) && count($ccs) && $this->getField("add_cc_to_observer")) {
          foreach ($ccs as $cc) {
             if ($cc != $requester
                && !Toolbox::inArrayCaseCompare($cc, $blacklisted_emails) // not blacklisted emails
-               && ($tmp = User::getOrImportByEmail($cc) > 0)) {
+               && (($tmp = User::getOrImportByEmail($cc)) > 0)) {
```
Comment (cedric-anne) [inline inc/mailcollector.class.php]:
```suggestion
               && ($tmp = User::getOrImportByEmail($cc)) > 0) {
```

File: inc/mailcollector.class.php [hunk 4]
```diff
@@ -970,7 +975,7 @@ function buildTicket($uid, \Zend\Mail\Storage\Message $message, $options = []) {
          foreach ($tos as $to) {
             if ($to != $requester
                && !Toolbox::inArrayCaseCompare($to, $blacklisted_emails) // not blacklisted emails
-               && ($tmp = User::getOrImportByEmail($to) > 0)) {
+               && (($tmp = User::getOrImportByEmail($to)) > 0)) {
```
Comment (cedric-anne) [inline inc/mailcollector.class.php]:
```suggestion
               && ($tmp = User::getOrImportByEmail($to)) > 0) {
```

File: install/update_94_95.php [hunk 5]
```diff
@@ -941,6 +941,10 @@ function update94to95() {
       }
    }
 
+     /** Add new option to mailcollector */
+     $migration->addField("glpi_mailcollectors", "add_cc_to_observer", "boolean");
+     /** /add new option to mailcollector */
```
Comment (cedric-anne) [inline install/update_94_95.php]: Indentation seems not correct.

Comment (stonebuzz) [reply install/update_94_95.php]: Fixed

File: inc/mailcollector.class.php (hunk 3 above)
Comment (stonebuzz) [reply inc/mailcollector.class.php]: revert to previous check version

File: inc/mailcollector.class.php (hunk 4 above)
Comment (stonebuzz) [reply inc/mailcollector.class.php]: revert to previous check version

Comment (stonebuzz) [reply inc/mailcollector.class.php]: this ```&& ($tmp = User::getOrImportByEmail($to) > 0)``` affected ```true``` value to ```$tmp``` variable while 

 ```&& (($tmp = User::getOrImportByEmail($to)) > 0))``` affected users_id value to ```$tmp``` variable

File: inc/mailcollector.class.php (hunk 3 above)
Comment (stonebuzz) [reply inc/mailcollector.class.php]: this ```&& ($tmp = User::getOrImportByEmail($to) > 0)``` affected ```true``` value to ```$tmp``` variable while 

 ```&& (($tmp = User::getOrImportByEmail($to)) > 0))``` affected users_id value to ```$tmp``` variable

File: inc/mailcollector.class.php (hunk 4 above)
Comment (stonebuzz) [reply inc/mailcollector.class.php]: At this point collector need users_id rather than succes (or not) when trying retrieve user

**Pull request merged**